docs/devel/kubectl-conventions.md

```diff
@@ -135,6 +135,77 @@ Updated: 8/27/2015
 * Use "TYPE" for the particular flavor of resource type accepted by kubectl, rather than "RESOURCE" or "KIND"
 * Use "NAME" for resource names
 
+## Command structure conventions
+
+For every command there should be a `NewCmd<CommandName>` function that creates the command and returns a pointer to a `cobra.Command`, which can later be added to other parent commands to compose the structure tree. There should also be a `<CommandName>Options` struct with a variable to every flag and argument declared by the command (and any other variable required for the command to run). This makes tests and mocking easier. The struct ideally exposes three functions:
+
+* `Complete`: Completes the struct variables with values that may not be directly provided, for example, by flags pointers. Here you will usually take the `args` slice and set the values as appropriate variables, instantiate configs or clients, etc.
```

Comment:
How is that? It's possible that I don't understand you but it seems to me that this structure respects* the separation between business logic (the functionality that the user cares about when running the command?) and the command framework (all the machinery that needs to be in place for the command to work?). 

[*] this is not 100% true because sometimes completing and validating may overlap eg. in [interactive commands](https://github.com/openshift/origin/pull/4433#discussion_r38985985) but still this structure is way better than a plain unit.